pkgs/by-name/op/opencode/package.nix

```diff
@@ -112,10 +111,30 @@ stdenvNoCC.mkDerivation (finalAttrs: {
   env.OPENCODE_VERSION = finalAttrs.version;
   env.OPENCODE_CHANNEL = "stable";
 
+  preBuild = ''
+    chmod -R u+w ./packages/opencode/node_modules
+    pushd ./packages/opencode/node_modules/@parcel/
+      for pkg in ../../../../node_modules/.bun/@parcel+watcher-*; do
+        linkName=$(basename "$pkg" | sed 's/@.*+\(.*\)@.*/\1/')
+        ln -sf "$pkg/node_modules/@parcel/$linkName" "$linkName"
+      done
+    popd
+
+    pushd ./packages/opencode/node_modules/@opentui/
+      for pkg in ../../../../node_modules/.bun/@opentui+core-*; do
+        linkName=$(basename "$pkg" | sed 's/@.*+\(.*\)@.*/\1/')
+        ln -sf "$pkg/node_modules/@opentui/$linkName" "$linkName"
+      done
+    popd
+  '';
+
   buildPhase = ''
     runHook preBuild
 
-    bun run ./script/build.ts --single
+
+    cd ./packages/opencode
+    cp ${./bundle.ts} ./bundle.ts
+    bun run ./bundle.ts
```

Comment:
Naive question: why not
```nix
bun run ${./bundle.ts}
```
?

Reply:
This doesn't work, since `bundle.ts` calculates paths and must be placed into the source directory. Else bun complains with

```
Cannot find module './node_modules/@opentui/solid/scripts/solid-plugin' from '/nix/store/i0g7pamyc83hl7n0k3vmfcn8ih35cdyk-bundle.ts'
```